### DISCUSSION TITLE
Chip CB2S / BK7231N replace to TYWE2S / ESP-02S or Flash ?

### DISCUSSION THREAD

**Discussion by ipn00tic** [arendst/Tasmota#17173] [Q&A]
Good afternoon friends

I bought GIRIER brand 16A smart plugs with CB2S / BK7231N Chip, is it better to replace with TYWE2S / ESP-02S chip and then is it easy to write Tasmota firmware or is it better to try to write OpenBK7231N/OpenBeken firmware in original chip?

Thanks

**Comment by barbudor** [comment]
I am a Tasmota user, so I would advise to transplant an ESP and use Tasmota
I have no idea on how those alternatives softwares behave comparatively to Tasmota

Now, the lazy way would be to 1st try thoses software without swapping the chips
And if you are ot happy with them, you know you can still do the transplant and lovely Tasmota on it 😍

**Reply by sfromis** [reply]
What I hear about OpenBK7231N is that it has a very few similarities with Tasmota, and by far not the features. Of course, your needs may be low.

If you decide to replace, as replacement for CB2S I can suggest:
https://templates.blakadder.com/WT32C3-01N.html (with ESP32, more features available)
https://templates.blakadder.com/ESP-02S.html (basic ESP8266)
https://templates.blakadder.com/WT-01N.html (basic ESP8266)

**Reply by marcelstoer** [reply]
The ESP-02S is actually powered by an ESP8285 to my knowledge. I haven't found any instructions how to flash it yet - let alone a programmer board.

**Reply by sfromis** [reply]
ESP8285 **is** an ESP8266, just coupled with flash. It is flashed as any other ESP8266, using tx/rx/vcc/gnd/io0. Note that IO0 has no pin on the edge, but a pad on the back side.
![image](https://github.com/arendst/Tasmota/assets/47082390/93d2ed96-3249-4b76-bb32-581980ed9a95)
